ocaml/xapi-guard/lib/varstored_interface.ml

```diff
@@ -102,8 +75,111 @@ let serve_forever_lwt rpc_fn path =
   Lwt_switch.add_hook (Some shutdown) cleanup ;
   Lwt.return cleanup
 
+let serve_forever_lwt_callback rpc_fn path _ req body =
+  let uri = Cohttp.Request.uri req in
+  match (Cohttp.Request.meth req, Uri.path uri) with
+  | `POST, _ ->
+      let* body = Cohttp_lwt.Body.to_string body in
+      let* response =
+        Dorpc.wrap_rpc err (fun () ->
+            let call = Xmlrpc.call_of_string body in
+            (* Do not log the request, it will contain NVRAM *)
+            D.debug "Received request on %s, method %s" path call.Rpc.name ;
+            rpc_fn call
+        )
+      in
+      let body = response |> Xmlrpc.string_of_response in
+      Cohttp_lwt_unix.Server.respond_string ~status:`OK ~body ()
+  | _, _ ->
+      let body =
+        "Not allowed"
+        |> Rpc.rpc_of_string
+        |> Rpc.failure
+        |> Xmlrpc.string_of_response
+      in
+      Cohttp_lwt_unix.Server.respond_string ~status:`Method_not_allowed ~body ()
+
+(* The TPM has 3 kinds of states *)
+type state = {
+    permall: string  (** permanent storage *)
+  ; savestate: string  (** for ACPI S3 *)
+  ; volatilestate: string  (** for snapshot/migration/etc. *)
+}
+
+let deserialize t =
+  match String.split_on_char ' ' t with
```

Comment:
The split character is shared between two functions. I would define a value for it.